packages/api/endpoints/token.js

```diff
@@ -1,124 +1,95 @@
-      }
-
-      const refresh = tokenInfo.refresh_token;
-      const userName = tokenInfo.endpoint.split('/').pop();
-      const expires = (+new Date()) + (tokenInfo.expires_in * 1000);
-
+    try {
+      const responseObject = await authHelpers.getToken(code);
+      const {
+        userName, accessToken, refresh, expires
+      } = responseObject;
       const u = new User();
 
       return u.get({ userName })
         .then(() => u.update({ userName }, { password: accessToken, refresh, expires }))
         .then(() => {
           if (state) {
+            log.info(`Log info: Redirecting to state: ${state} with token ${accessToken}`);
             return resp(context, null, 'Redirecting to the specified state', 301, {
               Location: `${decodeURIComponent(state)}?token=${accessToken}`
             });
           }
+          log.info('Log info: No state specified, responding 200');
           return resp(context, null, JSON.stringify({ token: accessToken }), 200);
-        }).catch((e) => {
-          log.error('User is not authorized', e);
+        })
+        .catch((e) => {
           if (e.message.includes('No record found for')) {
             return resp(context, new Error('User is not authorized to access this site'));
           }
           return resp(context, e);
         });
-    }).catch((e) => {
+    }
+    catch (e) {
       log.error('Error caught when checking code:', e);
-      resp(context, e);
-    });
+      return resp(context, e);
+    }
   }
   return resp(context, new Error('Request requires a code'));
 }
 
 /**
- * AWS API Gateway function that redirects to the correct URS endpoint with the correct client
- * ID to be used with the API
+ * `login` is an AWS API Gateway function that redirects to the correct
+ * authentication endpoint with the correct client ID to be used with the API
  *
- * @param {Object} event - aws lambda event object.
- * @param {Object} context - aws context object
- * @param {Function} cb - aws lambda callback function
- * @returns {Promise} the token or the callback object
+ * @param  {Object} event   - Lambda event object
+ * @param  {Object} context - Lambda context object
+ * @param  {Function} cb    - Lambda callback function
+ * @returns {Function}       Lambda callback function
  */
 function login(event, context, cb) {
-  const endpoint = process.env.EARTHDATA_BASE_URL;
-  const clientId = process.env.EARTHDATA_CLIENT_ID;
-
   const code = get(event, 'queryStringParameters.code');
   const state = get(event, 'queryStringParameters.state');
 
   if (code) {
     return token(event, context);
   }
 
-  let url = `${endpoint}/oauth/authorize?` +
-              `client_id=${clientId}&` +
-              `redirect_uri=${redirectUriParam()}&response_type=code`;
-  if (state) {
-    url = `${url}&state=${encodeURIComponent(state)}`;
-  }
+  const url = authHelpers.generateLoginUrl(state);
+
   return cb(null, {
```

Comment:
This is inconsistent return. if `code` is present returns a promise but otherwise it returns a function without a promise.